core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -65,8 +65,12 @@ object ConsoleConsumer extends Logging {
   def run(conf: ConsumerConfig) {
     val timeoutMs = if (conf.timeoutMs >= 0) conf.timeoutMs else Long.MaxValue
     val consumer = new KafkaConsumer(consumerProps(conf), new ByteArrayDeserializer, new ByteArrayDeserializer)
+
+    /*
+    Fix for KAFKA-6764. partition/offset is honoured only when group-id not passed.
+     */
     val consumerWrapper =
-      if (conf.partitionArg.isDefined)
+      if (!conf.groupIdPassed && conf.partitionArg.isDefined)
```

Comment:
should it be an error if groupId is passed along with partition/offset?

Reply:
It could be. But for consistency better to ignore. Because when groupId is passed, from-beginning is ignored if passed. So likewise.

Reply:
An error is thrown when group and partition is specified.